### PR TITLE
feat!: NEXT-39285 - allow overriding register task by passing an object

### PR DESCRIPTION
**Pull request by yusufttur**
- Adjust register task for commercial login
- Add a task to configure a sales channel

### CODE REVIEW

**Review by vanpham-sw — DISMISSED** [review]

File: src/tasks/shop-customer/Account/Register.ts [hunk 1]
```diff
@@ -18,53 +22,66 @@ export const Register = base.extend<{ Register: Task }, FixtureTypes>({
             company: 'shopware', 
             department: 'Operations', 
             vatRegNo: 'DE1234567890',
-        }
+        };
 
-        const task = (email: string, isCommercial?: boolean) => {
-            return async function Register() {
+        const registerTask = (overrides?: Partial<RegistrationData>, isCommercial?: boolean ) => {
```
Comment (frobel) [inline src/tasks/shop-customer/Account/Register.ts:27]: Why we change here the complete task?

**Review by frobel — DISMISSED** [review]

File: src/tasks/shop-customer/Account/Register.ts (hunk 1 above)
Comment (frobel) [inline src/tasks/shop-customer/Account/Register.ts:27]: Why we change here the complete task?

File: src/tasks/shop-admin/Settings/ConfigSalesChannel.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { test as base, expect } from '@playwright/test';
+import type { Task } from '../../../types/Task';
+import type { FixtureTypes} from '../../../types/FixtureTypes';
+
+export const ConfigSalesChannel = base.extend<{ ConfigSalesChannel: Task }, FixtureTypes>({
```
Comment (frobel) [inline src/tasks/shop-admin/Settings/ConfigSalesChannel.ts]: here you have to run the `clearCache` function because it is `systemConfig` related.

Comment (yusufttur) [reply src/tasks/shop-admin/Settings/ConfigSalesChannel.ts]: It works without cache clear.

**Review by frobel — APPROVED** [review]

**Pull request merged**